### PR TITLE
[meson] Update meson 0.55

### PR DESCRIPTION
**Pull request by Neumann-A**
and allow only the vcpkg internal version to be found and used on all platforms.
(since system packages of meson are probably too old/outdated) 

This unblocks a few follow up PRs. We should be able to crosscompile for arm-windows now.

### CODE REVIEW

**Comment by Neumann-A** [comment]
opend up https://github.com/mesonbuild/meson/issues/7443 for the CI regressions

**Comment by Voskrese** [comment]
don't really like that portable version is used python3.

**Comment by Voskrese** [comment]
>  --user      Install to the Python user install directory for your platform. Typically ~/.local/, or %APPDATA%\Python on
>              Windows. (See the Python documentation for site.USER_BASE for full details.)
> https://stackoverflow.com/questions/42988977/what-is-the-purpose-of-pip-install-user

https://github.com/microsoft/vcpkg/issues/12285

**Comment by Neumann-A** [comment]
@Voskrese: If it is on PATH and you forwarded your PATH to VCPKG or it is on a default search path for `find_program/find_file` vcpkg will not use its internal python.

File: scripts/azure-pipelines/linux/provision-image.sh [hunk 1]
```diff
@@ -27,7 +27,10 @@ APT_PACKAGES="$APT_PACKAGES libxkbcommon-dev libxcb-keysyms1-dev \
 APT_PACKAGES="$APT_PACKAGES libkrb5-dev"
 
 # Additionally required by mesa
-APT_PACKAGES="$APT_PACKAGES python3-setuptools"
+APT_PACKAGES="$APT_PACKAGES python3-setuptools python3-mako"
```
Comment (Neumann-A) [inline scripts/azure-pipelines/linux/provision-image.sh:30]: mako is required to actual build mesa in the future

**Comment by Neumann-A** [comment]
@BillyONeal: Requires VM update. (Or you tell me how I can check that python packages are actually installed without using pip ;))

**Comment by BillyONeal** [comment]
@Neumann-A Marked; can you add the provision-image.sh edit?

**Comment by Neumann-A** [comment]
@BillyONeal that is already done here https://github.com/microsoft/vcpkg/pull/12393/commits/813a020684fb86501b8a71b2b19b8b5a0d8f2dce

**Comment by BillyONeal** [comment]
@Neumann-A I am blind, thanks :)

**Comment by Voskrese** [comment]
> If it is on PATH and you forwarded your PATH to VCPKG or it is on a default search path for `find_program/find_file` vcpkg will not use its internal python.

obtained if PYTHON 3.5.2 and 3.7.3 are installed
will there be an error? all the USER folder is shared

**Comment by Neumann-A** [comment]
We should probably wait for 55.1 which solves the setuptools dependency. see #7461

**Comment by c72578** [comment]
Meson 0.55.1 has just been released.
https://github.com/mesonbuild/meson/releases/tag/0.55.1

**Comment by Neumann-A** [comment]
thx for the heads up. Currently finishing up my cmake compiler flags detection logic.

**Pull request closed without merging**